src/sentry/issues/status_change_consumer.py

```diff
@@ -106,27 +106,35 @@ def update_status(group: Group, status_change: StatusChangeMessageData) -> None:
         )
 
 
-def get_group_from_fingerprint(status_change: StatusChangeMessageData) -> Group | None:
-    grouphash = (
-        GroupHash.objects.filter(
-            project=status_change["project_id"],
-            hash=status_change["fingerprint"][0],
-        )
-        .select_related("group")
-        .first()
-    )
+def get_groups_from_fingerprints(
+    project_id: int, fingerprints: Iterable[Iterable[str]]
+) -> dict[str, Group]:
```

Comment:
I was thinking about a bulk api that looks more like

```python
def bulk_get_groups_from_fingerprints(
  fingerprints: Iterable[Tuple[int, Iterable[str]]]
) -> dict[Tuple[int, str], Group]:
```

where I can pass a list of project + fingerprint pairs as the groups may span across multiple projects.

Reply:
ISTR fingerprints are guaranteed unique at a project-level, so we'd have to split up and query GroupHash by project in that case. @wedamija is that correct?

We could iterate over the set of projects and query GroupHash once per project in the method, but not sure what the best practice here is. It might make more sense for the caller to call once per project instead.